src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/external_repacking_adjuster.cpp

```diff
@@ -51,14 +51,24 @@ VectorDims BrgemmExternalRepackingAdjuster::get_blk_order(size_t shape_rank) {
     return order;
 }
 
-VectorDims BrgemmExternalRepackingAdjuster::get_blk_shape(const VectorDims& planar_shape, ov::element::Type prc) {
-    const auto vnni_factor = brgemm_utils::compute_vnni_factor(prc);
+VectorDims BrgemmExternalRepackingAdjuster::get_blk_shape(const VectorDims& planar_shape,
+                                                          ov::element::Type prc,
+                                                          bool is_transposed) {
+    size_t block = 1;
+    if (is_transposed) {
+        // In case of transpose, K dimension must be rounded-up to number of elems in vector register
+        // For the details, please see 'transpose16x8' and 'fixup16x16' implementations and usage in
+        // onednn/src/cpu/x64/matmul/brgemm_matmul_copy_utils.cpp
+        block = brgemm_utils::get_elems_in_vec(prc);
+    } else {
+        block = brgemm_utils::compute_vnni_factor(prc);
+    }
```

Comment:
If I understand correctly, we have the same logic in `RepackedWeightsBufferExpression::init_allocation_size`. Maybe we can create a separate brgemm_utils helper which computes K_blk based on precision and layout? What do you thinks?

Reply:
Good point. Created helper `compute_buffer_b_allocation_shape`. Thanks!